expression/like.go

```diff
@@ -129,18 +135,23 @@ func compilePattern(pattern string) (patChars, patTypes []byte) {
 		var tp byte
 		var c = pattern[i]
 		switch c {
-		case '\\':
+		case escape:
 			lastAny = false
 			tp = patMatch
 			if i < len(pattern)-1 {
 				i++
 				c = pattern[i]
-				if c == '\\' || c == '_' || c == '%' {
+				if c == escape || c == '_' || c == '%' {
 					// valid escape.
 				} else {
-					// invalid escape, fall back to literal back slash
+					// invalid escape, fall back to escape byte
+					// mysql will treat escape character as the origin value even
+					// the escape sequence is invalid in Go or C.
+					// e.g, \m is invalid in Go, but in MySQL we will get "m" for select '\m'.
+					// Following case is correct just for escape \, not for others like +.
+					// TODO: add more checks for other escapes.
 					i--
-					c = '\\'
+					c = escape
 				}
 			}
```

Comment:
I just use the old code here, which just seems using the origin escape character directly. 
I have added a comment that we will add more tests later.